docs/user-guide/grpc.md

```diff
@@ -0,0 +1,148 @@
+# gRPC example
+
+This section explains how to use Traefik as reverse proxy for gRPC application with self-signed certificates.
+
+!!! warning
+    As gRPC needs HTTP2, we need valid HTTPS certificates on both gRPC Server and Træfik.
+
+<p align="center">
+<img src="/img/grpc.svg" alt="gRPC architecture" title="gRPC architecture" />
+</p>
+
+## gRPC Server certificate
+
+In order to secure the gRPC server, we generate a self-signed certificate for backend url:
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout ./backend.key -out ./backend.crt
+```
+
+That will prompt for information, the important answer is:
+
+```
+Common Name (e.g. server FQDN or YOUR name) []: backend.local
+```
+
+## gRPC Client certificate
+
+Generate your self-signed certificate for frontend url:
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout ./frontend.key -out ./frontend.crt
+```
+
+with
+
+```
+Common Name (e.g. server FQDN or YOUR name) []: frontend.local
+```
+
+## Træfik configuration
+
+At last, we configure our Træfik instance to use the both self-signed certificate.
```

Comment:
Little grammar improvement:

_use the both self-signed certificate_ -> _use both self-signed certificates_